src/character.cpp

```diff
@@ -2645,3 +2647,58 @@ float Character::healing_rate( float at_rest_quality ) const
 
     return final_rate;
 }
+
+float Character::healing_rate_medicine( float at_rest_quality, const body_part bp ) const
+{
+    float rate_medicine = 0.0f;
+    float bandaged_rate = 0.0f;
+    float disinfected_rate = 0.0f;
+
+    if( has_effect( effect_bandaged, bp ) ) {
+        const effect &e_bandaged = get_effect( effect_bandaged, bp );
+        bandaged_rate += e_bandaged.get_amount( "HEAL_RATE", 0 ) / HOURS( 24 );
+
+        if( bp == bp_head ) {
+            bandaged_rate *= e_bandaged.get_amount( "HEAL_HEAD_MULT", 0 ) / 100.0f;
+        }
+        if( bp == bp_torso ) {
+            bandaged_rate *= e_bandaged.get_amount( "HEAL_TORSO_MULT", 0 ) / 100.0f;
+        }
+    }
+
+    if( has_effect( effect_disinfected, bp ) ) {
+        const effect &e_disinfected = get_effect( effect_disinfected, bp );
+        disinfected_rate += e_disinfected.get_amount( "HEAL_RATE", 0 ) / HOURS( 24 );
+
+        if( bp == bp_head ) {
+            disinfected_rate *= e_disinfected.get_amount( "HEAL_HEAD", 0 ) / 100.0f;
+        }
+        if( bp == bp_torso ) {
+            disinfected_rate *= e_disinfected.get_amount( "HEAL_TORSO", 0 ) / 100.0f;
+        }
+    }
+
+    rate_medicine += bandaged_rate + disinfected_rate;
+    rate_medicine *= 1.0f + mutation_value( "healing_resting" );
+
+    if( at_rest_quality > 0.0f ) {
+            bandaged_rate *= 2;
```

Comment:
Would be more future-proof if you had it scale with rest quality. Say, just add it to bandaged rate.
Why does it only affect bandages and not disinfection?

Reply:
It was a bug, "at_rest_quality" must be applyed to all healing.